### ISSUE TITLE
Null values when reusing sso token in expo-apple-authentication

### ISSUE THREAD

**Issue by samelung** [expo/expo#6538]
### Problem
We don't know if this is a bug or is working as designed. As we understand it, sign in with apple only gives user information the first time an app is installed, and any subsequent attempt the return values will be NULL... which Expo mentions in the documentation.

However what happens if the app needs to be reinstalled for any reason? Sign In With Apple "see" it as a simple login to the app and return NULL. Because of the Delete and reinstall all metadata regarding the app has been deleted and when Sign in with Apple returns NULL. Then it is impossible to map the user to the users data in the backend. 

Getting around the problem; is asking the end user to delete the Sign in token installed under settings and Apps Using Your Apple ID. For many endusers this is too advanced for them to remember and do.

The question is: Are we doing something wrong or how do we get around this issue ?

Please advice

Best regards

The below is a simple work through for testing the issue.

### Environment
This issue occurs when using a standalone app.
Issue happens only when using IOS 13.
Issue happens only when a SSO token is being reused.

### Steps to Reproduce

We hope somebody can advice. We don't know if this is expected behavior when using sign in with apple or it is a bug.

But this is what happens :

     1) Install app first time 
     2) Sign in with apple

NOTE: Everything works as expected and a token is being stored in settings under "Apps Using Your Apple ID"

     3) Delete app
     4) Reinstall app
     5) When login again the phone reuses the token under settings and ask the user if 
        he/she wants to sign in with an existing token on device.

NOTE: Behavior registered:

     values in 
          AppleAuthentication.AppleAuthenticationScope.FULL_NAME
          AppleAuthentication.AppleAuthenticationScope.EMAIL
     is set to NULL. Which is not expected. We expected Full_Name and Email was fulfilled.

     6) Not able to connect the user to the user account because of Null values.

The "Work Around":

     7) Delete the app from device
     8) Delete token in settings under "Apps Using Your Apple ID"
     9) Install app again
     10) Register using sign in with apple

NOTE: Everything works as expected. But how do we explain end users to delete the token first before they reinstall the app again ?

Please advice

**Comment by cruzach** [comment]
One option is to store this information outside of the app (server-side), for example in your user database if you have one. Another option is [securestore](https://docs.expo.io/versions/latest/sdk/securestore/) which uses keychain services under the hood, and as long as the app is signed with the same provisioning profile, the app will have access to data set in the previous install

**Comment by cruzach** [comment]
Hey @samelung - it looks like we don't have any control over this, so it is the expected behavior. Best route is to use the suggestions above, so I'm going to close this issue

**Comment by samelung** [comment]
Hi, 

Thanks for the fast response, and sorry Im late on answer. We found a solution within 'sign in with apple'.

Thanks for the advice using secure store it is a smart solution to solve the issue saving information between reinstalls or logins.

However there is a solutions which we did not catch in the documentation, and maybe it should be rewritten a little bit.

In the response from 'credentials = AppleAuthentication.signInAsync({...', the response credentials.user is a unique key identifying the appleuser.

If the credentials.user is being saved either in securestore or in a backend database the problem has been solved.

Again thank you very much for the help in this matter!
